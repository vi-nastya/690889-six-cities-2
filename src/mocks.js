export const MOCK_CITIES = [{
  name: `City1`,
  location: {
    latitude: 52.0,
    longitude: 4.5,
    zoom: 10
  }
},
{
  name: `City2`,
  location: {
    latitude: 20.4,
    longitude: 10.1,
    zoom: 3
  }
}
];

export const MOCK_USER = {
  id: 4,
  isPro: false,
  name: `Max`,
  avatarUrl: `img/1.png`
};

export const MOCK_OFFERS = [{
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  maxAdults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];

export const MOCK_OFFER_UPDATED_FAVORITE = {
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
};

export const MOCK_OFFERS_UPDATED_FAVORITE = [{
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  maxAdults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];


export const MOCK_REVIEWS = [{
  id: 1,
  user: MOCK_USER,
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
}];

export const MOCK_AUTH = {
  id: 4,
  isPro: false,
  name: `Max`,
  email: `max@test.com`,
  avatarUrl: `img/1.png`
};


/* eslint-disable camelcase */
export const MOCK_AUTH_SERVER = {
  id: 4,
  is_pro: false,
  name: `Max`,
  email: `max@test.com`,
  avatar_url: `img/1.png`
};


export const MOCK_USER_SERVER = {
  id: 4,
  is_pro: false,
  name: `Max`,
  avatar_url: `img/1.png`
};

export const MOCK_REVIEWS_SERVER = [{
  id: 1,
  user: MOCK_USER_SERVER,
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
}];


export const MOCK_OFFERS_SERVER = [{
  id: 1,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: false,
  is_premium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: true,
  is_premium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  max_adults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: false,
  is_premium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];

export const MOCK_OFFER_UPDATED_FAVORITE_SERVER = {
  id: 1,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: true,
  is_premium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
};
/* eslint-enable */
