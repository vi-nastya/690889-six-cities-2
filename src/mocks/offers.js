export const offers = [
  {
    id: 0,
    title: `Wood and stone place`,
    price: 80,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.3909553943508, longitude: 4.85309666406198},
    priceText: `night`,
    type: `Private room`,
    picture: `img/apartment-01.jpg`,
    isPremium: false,
    isFavorite: true,
    images: [`img/1.png`, `img/2.png`],
    bedrooms: 2,
    rating: 4,
    maxAdults: 2,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      is–óro: true,
      name: `Angelina`,
      avatarUrl: `img/1.png`
    },
  },
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    price: 120,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.369553943508, longitude: 4.85309666406198},
    priceText: `night`,
    type: `Apartment`,
    picture: `img/apartment-02.jpg`,
    isPremium: true,
    isFavorite: false,
    rating: 3.6,
    images: [``]
  },
  {
    id: 2,
    title: `Nice, cozy, warm big bed apartment`,
    price: 180,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.3909553943508, longitude: 4.929309666406198},
    priceText: `night`,
    type: `Apartment`,
    picture: `img/apartment-03.jpg`,
    isPremium: true,
    isFavorite: true,
    images: [``],
    rating: 2,
  },
  {
    id: 3,
    title: `Wood and stone place`,
    price: 90,
    city: {name: `Moscow`, location: {latitude: 52.3809553943508, longitude: 4.939309666406198}},
    location: {latitude: 52.3809553943508, longitude: 4.939309666406198},
    priceText: `night`,
    type: `Private room`,
    picture: `https://picsum.photos/260/200`,
    isPremium: false,
    isFavorite: false,
    images: [``],
    rating: 4.8,
  },
];
