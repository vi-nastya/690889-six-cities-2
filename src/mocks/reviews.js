export const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      isPro: true,
      name: `Mark`,
      avatarUrl: `img/2.png`
    },
    rating: 2,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-10T14:13:56.569Z`
  }
];
