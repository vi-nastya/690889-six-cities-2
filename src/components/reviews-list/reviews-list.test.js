import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list";
import {reviews} from "../../mocks/reviews";

it(`ReviewsList component is rendered correctly after relaunch`, () => {
  const tree = renderer.create(<ReviewsList reviewsData={reviews}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
