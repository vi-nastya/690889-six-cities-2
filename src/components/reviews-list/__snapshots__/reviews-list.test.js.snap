// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList component is rendered correctly after relaunch 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/1.png"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        May 2019
      </time>
    </div>
  </li>
</ul>
`;
